python/pyspark/sql/functions.py

```diff
@@ -10403,6 +10405,82 @@ def udf(
         return _create_py_udf(f=f, returnType=returnType, useArrow=useArrow)
 
 
+def udtf(
+    f: Optional[Type] = None,
+    returnType: Union[StructType, str] = None,
```

Comment:
If you'd like to require this always invoked via `udf(returnType="schema")` you can fix it as below:

```suggestion
    *,
    returnType: Union[StructType, str],
```

which makes sense per the type hints.

If you'd like to allow a usage such as `@udf("schema")`, the type hint of `f` has to be fixed too.

We can just go ahead merge first, and I can followup with fixing the type hints.

Reply:
I think it's more readable if we require this to be invoked like `@udtf(returnType="...")`, and it doesn't make sense for UDTFs to have a default value for the return type (unlike udfs which use `StringType`).